libraries/support/azure/graph.rb

```diff
@@ -0,0 +1,66 @@
+# frozen_string_literal: true
+
+require 'json'
+require 'singleton'
+
+module Azure
+  class Graph
+    include Singleton
+
+    def with_client(azure_client, override: false)
+      set_reader(:azure_client, azure_client, override)
+    end
+
+    def for_tenant(tenant_id, override: false)
+      set_reader(:tenant_id, tenant_id, override)
+    end
+
+    def get_user(object_id)
+      get(
+          "/#{tenant_id}/users/#{object_id}",
+          params: { 'api-version' => '1.6' }
+      )
+    end
+
+    def get_users
+      get(
+          "/#{tenant_id}/users",
+          params: { 'api-version' => '1.6' }
+      )
+    end
+
+    def get_users_next(next_page)
+      get(
+          "/#{tenant_id}/#{next_page}",
+          params: { 'api-version' => '1.6' }
+      )
+    end
+
+    private
+
+    def get(*args)
+      confirm_configured!
+      response          = azure_client.get(*args).body
+      map               = {}
+      map["values"]     = response.fetch('value', response)
+      map["nextLink"] ||= response.fetch("odata.nextLink", nil)
+      map
+    end
```

Comment:
Agree, this is perhaps leaving too much to dev knowledge. It is linked to the "where should we paginate" you mentioned above and so similarly I will wait until we have discussed further.

>the get definition should also line up with what's already in Management. (Keyword arguments :url, :api_version.)
I have seen in some of the MS documentation that there are places where params other than `api_version` would be very beneficial (e.g. providing a filter to only get certain entities, instead of getting all entities and hand rolling the logic), is there a reason not to allow a caller to provide their own hash of params instead of limiting to only `api-version`?